src/main/java/org/mitre/synthea/modules/DeathModule.java

```diff
@@ -28,7 +28,16 @@
    * @param time - the time of the death exam and certification.
    */
   public static void process(Person person, long time) {
-    if (!person.alive(time) && person.attributes.containsKey(Person.CAUSE_OF_DEATH)) {
+    if (person.alive(time)) {
+      // remove the death from the record.
+      // otherwise it's tough to tell after the fact whether this patient did die or will die
+      person.attributes.remove(Person.CAUSE_OF_DEATH);
+      person.record.death = null;
+      if (person.hasMultipleRecords) {
+        person.records.values().forEach(hr -> hr.death = null);
+      }
+      // TODO: if stopping/restarting/reloading becomes a thing, this may have to be reworked
+    } else if (person.attributes.containsKey(Person.CAUSE_OF_DEATH)) {
```

Comment:
I don't understand why you are changing this code.

If the person is alive right now, you are removing their cause of death? If you do that, in the future, when they actually die, we won't know why. (in the save and restore scenario that @hadleynet is working on)

And it doesn't even look like you are examining the cause of death attribute in the prevalence report anyway...

One thought is that in the prevalence report, instead of SQL like `WHERE p.DATE_OF_DEATH is null` you could do something like `WHERE p.DATE_OF_DEATH is null or p.DATE_OF_DEATH > ?` and you set the date to when the simulation stopped (which could be passed in with the Generator)

Reply:
Yeah that SQL is the reason why this section was modified. The trouble with doing `where DATE_OF_DEATH > ?`  is that the DB doesn't contain any info on when the simulation stopped so there's no way to tell after the fact which patients with a death date were actually still alive at the end of the simulation. Not that the DB is a majorly used feature but being able to tell which patients are alive and which are dead is an important distinction.

Reply:
I know, but the exporters do know, and you could pass that information into the PrevalenceReport exporter.

Reply:
I'm thinking of the file-based DB option where the patient records are persistent but the end date of the simulation isn't. That said I don't know if anyone actually uses that feature.
I had forgotten that all the other exporters reference the simulation stop time to check alive/dead status so I can do the same thing here.

Maybe this would be a separate PR but what if I added a new table in the DB that tracked the details of the simulation? It would include things like the seed, the simulation stop time, maybe some run settings, and an ID to link back to the patients and providers from that run.

Reply:
FWIW, in the serialization-support branch, I added a "lastUpdated" field to Person to track the end of each simulation run so I can restart where it stopped after a serialize/deserialize.

Reply:
Reverted this section and updated the report SQL with `or p.DATE_OF_DEATH > ?`